internal/pkg/deploy/cloudformation/stack/scheduled_job.go

```diff
@@ -0,0 +1,300 @@
+	case strings.HasPrefix(schedule, daily):
+		return fmt.Sprintf(fmtCronScheduleExpression, cronDaily), nil
+	case strings.HasPrefix(schedule, weekly):
+		return fmt.Sprintf(fmtCronScheduleExpression, cronWeekly), nil
+	case strings.HasPrefix(schedule, monthly):
+		return fmt.Sprintf(fmtCronScheduleExpression, cronMonthly), nil
+	case strings.HasPrefix(schedule, annually):
+		fallthrough
+	case strings.HasPrefix(schedule, yearly):
+		return fmt.Sprintf(fmtCronScheduleExpression, cronYearly), nil
+	default:
+		return "", fmt.Errorf("unrecognized cron definition string")
+	}
+}
+
+func specified(input string) bool {
+	return !strings.ContainsAny(input, "*?")
+}
+
+// toAWSCron converts "standard" 5-element crons into the AWS preferred syntax
+// cron(* * * * ? *)
+// MIN HOU DOM MON DOW YEA
+// EITHER DOM or DOW must be specified as ? (either-or operator)
+// BOTH DOM and DOW cannot be specified
+// DOW numbers run 1-7, not 0-6
+func toAWSCron(schedule string) (string, error) {
+	const (
+		MIN = iota
+		HOU
+		DOM
+		MON
+		DOW
+	)
+
+	// Split the cron into its components. We can do this because it'll already have been validated.
+	// Use https://golang.org/pkg/strings/#Fields since it handles consecutive whitespace.
+	sched := strings.Fields(schedule)
+
+	// Check whether the Day of Week and Day of Month fields have a ?
+	// Possible conversion:
+	// * * * * * ==> * * * * ?
+	// 0 9 * * 1 ==> 0 9 ? * 1
+	// 0 9 1 * * ==> 0 9 1 * ?
+	switch {
+	// If both are unspecified, convert DOW to a ? and DOM to *
+	case !specified(sched[DOM]) && !specified(sched[DOW]):
+		sched[DOW] = "?"
+		sched[DOM] = "*"
+	// If DOM is * or ? and DOW is specified, convert DOM to ?
+	case !specified(sched[DOM]) && specified(sched[DOW]):
+		sched[DOM] = "?"
+	// If DOW is * or ? and DOM is specified, convert DOW to ?
+	case !specified(sched[DOW]) && specified(sched[DOM]):
+		sched[DOW] = "?"
+	// Error if both DOM and DOW are specified
+	default:
+		return "", errors.New("cannot specify both DOW and DOM in cron expression")
+	}
+
+	// Increment the DOW by one if specified as a number
+	// https://play.golang.org/p/_1uxt0zJneb
+	var newDOW []rune
+	for _, c := range sched[DOW] {
+		if unicode.IsDigit(c) {
+			newDOW = append(newDOW, c+1)
+		} else {
+			newDOW = append(newDOW, c)
+		}
+	}
+	// We don't need to use a string builder here because this will only ever have a max of
+	// about 50 characters (SUN-MON,MON-TUE,TUE-WED,... is the longest possible string here)
+	sched[DOW] = string(newDOW)
+
+	// Add "every year" to 5-element crons to comply with AWS
+	sched = append(sched, "*")
+
+	return fmt.Sprintf(fmtCronScheduleExpression, strings.Join(sched, " ")), nil
+}
+
+// StateMachine converts the Timeout and Retries fields to an instance of template.StateMachineOpts
+// It also performs basic validations to provide a fast feedback loop to the customer.
+func (j *ScheduledJob) stateMachine() (*template.StateMachineOpts, error) {
+
```

Comment:
Since it essentially generates the state machine option.
```suggestion
func (j *ScheduledJob) stateMachineOpts() (*template.StateMachineOpts, error) {
```